charts/tfy-karpenter-config/files/al2023-soci-gpu-provisioner-userdata.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+set -ex
+
+version_lte() {
+  printf '%s\n' "$1" "$2" | sort -C -V
+}
+
+version_lt() {
+  ! version_lte "$2" "$1"
+}
+
+CONTAINERD_VERSION_OUTPUT="$(containerd --version)"
+IFS=" " read -a  CONTAINERD_VERSION_STRING <<< "${CONTAINERD_VERSION_OUTPUT}"
+CONTAINERD_VERSION=${CONTAINERD_VERSION_STRING[2]}
+SANDBOX_IMAGE=$(containerd config dump | grep -oE 'sandbox_image = "(.*)"' | sed 's/sandbox_image = //')
+KUBELET_CONFIG_DIR="/etc/kubernetes/kubelet/config.json.d"
+SOCI_KUBELET_CONFIG_PATH="$KUBELET_CONFIG_DIR/99-soci.conf"
+CONTAINERD_CONFIG_FILEPATH="/etc/containerd/config.toml"
+BACKUP_CONTAINERD_CONFIG_FILEPATH="$CONTAINERD_CONFIG_FILEPATH.bak"
+SOCI_RELEASE_VERSION="0.7.0"
```

Comment:
Should we try out the version 0.8..0

Reply:
> Should we try out the version 0.8..0

We should pick this up seperatly. Let us release this first.